### PR TITLE
Add `pad_to_multiple_of` to GRPOTrainer and RLOOTrainer

### PR DESCRIPTION
**Pull request by czkkkkkk**
This PR adds `pad_training_to_multiple_of` to GRPOConfig, which pads both `prompt_ids` and `completion_ids` after rollout to a multiple of a specified number. This reduces the number of different `input_ids` shapes during training, minimizing recompilation for backends with static shape computation such as Neuron.

Noted that: this PR also offloads the padding of `prompt_ids` and `completion_ids` to CPU. The padding for them are not separated since their padded tensors are used later.

# What does this PR do?

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: trl/trainer/grpo_trainer.py [hunk 1]
```diff
@@ -1640,22 +1640,49 @@ def _generate_and_score_completions(
         ) = self._generate(prompts)
 
         # Convert lists of token IDs to padded tensors
-        prompt_ids = [torch.tensor(ids, device=device) for ids in prompt_ids_list]
```
Comment (qgallouedec) [inline trl/trainer/grpo_trainer.py:1646]: is it important to pad before sending to the device?

Comment (czkkkkkk) [reply trl/trainer/grpo_trainer.py:1646]: Yes, it is important for Neuron to avoid recompilation. If we pad after sending to the device, the input shape of the padding operator is changing across steps, which triggers recompilation.

File: trl/trainer/grpo_config.py [hunk 2]
```diff
@@ -461,6 +461,10 @@ class GRPOConfig(BaseConfig):
         default=None,
         metadata={"help": "Implementation of the cache method for faster generation when use_vllm is set to False."},
     )
+    pad_training_to_multiple_of: int | None = field(
+        default=None,
+        metadata={"help": "If set, pad the training sequences to a multiple of this value."},
```
Comment (michaelbenayoun) [inline trl/trainer/grpo_config.py]: Maybe specify here that **each** sequence (e.g. both the prompt_ids and the completion_ids) is padded to a multiple to this value.

Comment (czkkkkkk) [reply trl/trainer/grpo_config.py]: Make sense. I changed the description to "If set, pad both prompt_ids and completion_ids to a multiple of this value before training."

**Comment by qgallouedec** [comment]
Thanks, I've

- extended the same logic to the others trainers for consistency
- renamed `pad_training_to_multiple_of` to `pad_to_multiple_of`, for consistency with the rest of the codebase again
- added the description the the configs docstring

I'll just add a test and we're good to merge

**Comment by qgallouedec** [comment]
@codex review

**Comment by qgallouedec** [comment]
@codex review

**Comment by qgallouedec** [comment]
@codex review
